ee/tabby-webserver/src/service/thread.rs

```diff
@@ -52,13 +52,29 @@ impl ThreadService for ThreadServiceImpl {
             .next())
     }
 
+    async fn get_thread_message(&self, id: &ID) -> Result<Message> {
+        Ok(self
+            .db
+            .get_thread_message(id.as_rowid()?)
+            .await?
+            .ok_or_else(|| CoreError::NotFound("Message not found"))?
+            .try_into()?)
+    }
+
     async fn set_persisted(&self, id: &ID) -> Result<()> {
         self.db
             .update_thread_ephemeral(id.as_rowid()?, false)
             .await?;
         Ok(())
     }
 
+    async fn update_thread_message(&self, input: &UpdateMessageInput) -> Result<()> {
+        self.db
+            .append_thread_message_content(input.id.as_rowid()?, &input.content)
+            .await?;
+        Ok(())
```

Comment:
1. the db function need to take both thread_id and id to ensure id is part of thread_id.
2. update_thread_message_content, not append.